protocol/binary/envelope.go

```diff
@@ -52,6 +51,24 @@ func (bw *Writer) WriteEnveloped(e wire.Envelope) error {
 	return bw.WriteValue(e.Value)
 }
 
+// WriteLegacyEnveloped writes enveloped value using the non-strict envelope
+// (non-strict lacks an envelope version).
+func (bw *Writer) WriteLegacyEnveloped(e wire.Envelope) error {
```

Comment:
should we export this? can we just keep it private since we're pretty confident there isn't a good reason for users to use this?

Reply:
In this case, we’re using protocol/binary.Writer from the protocol package, so this does need to be public. We might have had the foresight to put this in the same package or an internal package, but sadly missed that opportunity.

Reply:
I don't feel strongly about this but as discussed offline,
WriteLegacyEnveloped isn't necessary because protocol implementations are
expected to be able to decode both, strict and non-strict envelopes regardless
of what they're sending.

Looking at the Apache Thrift Python library implementation, non-strict
envelopes are rejected only if the user explicitly opted into it. Strict
envelopes (which is what ThriftRW Go uses) are never rejected, even if the
user opted into writing non-strict envelopes.

Reply:
I buy it.

Reply:
The logic to detect the type of envelope (if any) seems very tied to the `binary` protocol -- should all of that logic live inside of the `binary` package? That way `WriteLegacyEnvelope` is not exported, but we still support it for this specific use case.